networkit/cpp/embedding/BiasedRandomWalk.hpp

```diff
@@ -0,0 +1,41 @@
+/*
+ * BiasedRandomWalk.hpp
+ *
+ *  Created on: 03.07.2020
+ *      Author: Klaus Ahrens  <ahrens@informatik.hu-berlin.de>
+ *
+ *  adapted and reimplemented from node2vec
+ *  part of snap [https://github.com/snap-stanford/snap]
+ *  Copyright (c) 2007-2019, Jure Leskovec (under BSD license)
+ *
+ *  see [https://arxiv.org/pdf/1607.00653v1.pdf]
+ *
+ */
+
+// networkit-format
+
+#ifndef BIASEDRANDOMWALK_HPP
+#define BIASEDRANDOMWALK_HPP
+
+#include <iostream>
+#include <vector>
+
+#include <networkit/graph/Graph.hpp>
+
+namespace NetworKit {
+namespace Embedding {
+
+using Walk = std::vector<node>;     // one walk
+using AllWalks = std::vector<Walk>; // n walks for all nodes: 2 dimensions in one big chunk
+
+/// preprocesses transition probabilities for random walks. Has to be called once before doWalks
+/// calls
+void preprocessTransitionProbs(const Graph &graph, double paramP, double paramQ);
+
+/// Simulates walks from every node and writes it into walks vector
+AllWalks doWalks(const Graph &graph, count walkLen, count numberOfWalks);
+
```

Comment:
A way to avoid defining global variables is to create `BiasedRandomWalk` class here.